src/main/kotlin/org/jitsi/nlj/transform/node/Node.kt

```diff
@@ -214,19 +218,123 @@ sealed class StatsKeepingNode(name: String): Node(name) {
      * other nodes, so that [Node] can keep track of its statistics.
      */
     protected fun doneProcessing(packetInfos: List<PacketInfo>) {
-        val processingDuration = System.nanoTime() - startTime
-        totalProcessingDuration += processingDuration
-
-        numOutputPackets += packetInfos.size
-        packetInfos.forEach {
-            it.addEvent(nodeExitString)
+        if (enableStatistics) {
+            val processingDuration = System.nanoTime() - startTime
+            stats.totalProcessingDurationNs += processingDuration
+            stats.maxProcessingDurationNs = Math.max(stats.maxProcessingDurationNs, processingDuration)
+
+            stats.numOutputPackets += packetInfos.size
+            packetInfos.forEach {
+                it.addEvent(nodeExitString)
+            }
         }
     }
 
     protected fun packetDiscarded(packetInfo: PacketInfo) {
-        numDiscardedPackets++
+        stats.numDiscardedPackets++
         BufferPool.returnBuffer(packetInfo.packet.buffer)
     }
+
+
+    override fun stop() {
+        if (stopped) {
+            return
+        }
+        stopped = true
+
+        if (enableStatistics && stats.numInputPackets > 0) {
```

Comment:
might we still want to know stats if it received nothing (to confirm where packets never made it, maybe)? or maybe that's still obvious by the lack of stats

Reply:
These are aggregates, so we wouldn't really see the missing packets. I added the check because health checks result in Nodes being created and they kept bumping the aggregate count (but nothing else, since nothing flows through them)